### PR TITLE
Add options for `extract-tests` to reuse existing versions or ignore new dependencies

### PR DESCRIPTION
**Pull request by simonihmig**
This is an attempt to implement the use cases of #64.

`--reuse-existing-versions` will replace any version specifier of the test-app with the one that existed before the migration, e.g. a pinned one. 

`--ignore-new-dependencies` will remove dependencies from test-app if they weren't in use before. This is admittedly a high risk option, in case we remove things that are not optional and do not come from "elsewhere" (like centralized linting configs), but it worked out quite ok for us...

No test coverage here, idk if you like to see some for these flags? Would need to be some fixture-based tests that actually assert file contents instead of "just" doing some smoke-tests that the migration results passes linting/tests, right? (Tbh, I wouldn't mind not having to do those, at least at this stage of this project... 🙃)

/cc @nicolechung

### CODE REVIEW

**Comment by NullVoxPopuli** [comment]
Thanks for doing this!

> fixture-based tests 
> actually assert file contents

Yeah, i think you're right that we probably don't want to run install, lint, test, etc, like the other tests.
I think maybe (?) Similar setup from the other tests could be used to test these new flags and assert the contents of the package.json (the fixtures folder maybe could get away with minimal things?)
Maybe there should be a separate folder for non-smoke fixtures for these kinds of 'run the tool, check the file contents' tests? (Iirc, the smoke tests iterate over the existing fixtures folder contents)
That could need some setup work, and is totally fine to do later.

**Review by NullVoxPopuli — APPROVED** [review]

File: cli/bin.js [hunk 1]
```diff
@@ -90,6 +90,18 @@ yarg
         type: 'boolean',
         default: false,
       });
+      yargs.option('reuse-existing-versions', {
```
Comment (nicolechung) [inline cli/bin.js:93]: I guess this is what makes it show up in `--help`? If so disregard question #2 from earlier....

Comment (simonihmig) [reply cli/bin.js:93]: Yeah, haven't tested tbh, but I think the `yargs` package should do this ootb

Comment (NullVoxPopuli) [reply cli/bin.js:93]: it does

**Pull request merged**